bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java

```diff
@@ -62,12 +64,29 @@ public SomfyTahomaBaseThingHandler(Thing thing) {
 
     @Override
     public void initialize() {
+        if (getThing().getProperties().containsKey(RSSI_LEVEL_STATE)) {
+            createRSSIChannel();
+        }
         updateStatus(ThingStatus.ONLINE);
     }
 
+    private void createRSSIChannel() {
+        if (thing.getChannel(RSSI) == null) {
+            logger.debug("{} Creating a rssi channel", getURL());
+            createChannel(RSSI, "Number","RSSI Level");
+        }
+    }
+
+    private void createChannel(String name, String type, String label) {
+        ThingBuilder thingBuilder = editThing();
+        Channel channel = ChannelBuilder.create(new ChannelUID(thing.getUID(), name), type).withLabel(label).build();
+        thingBuilder.withChannel(channel);
+        updateThing(thingBuilder.build());
+    }
+
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
-        logger.debug("Received command {} for channel {}", command, channelUID);
+        logger.debug("{} Received command {} for channel {}", getURL(), command, channelUID);
```

Comment:
`getURL` is quite expensive and called even if the log level is lower than DEBUG. Is there any reason why the URL is not set during initialization and stored in a String field? In case of configuration change the thing handler is disposed and re-initialized, so this should not be a problem. (also applies below)

Reply:
nice hint! fixed